File: findJudge.js
const findJudge = (n, trust) => {
  let notJudge = {};
  for (let relationship of trust) {
    notJudge[relationship[1]] = [...notJudge[relationship1], relationship[0]];
  }
  console.log(notJudge);

  if (notJudge.length === n - 1) {
    console.log(notJudge);
    for (let i = 1; i <= n; i++) {
      if (!(i in notJudge)) {
        return i;
      }
    }
  }
  return -1;
};

console.log(
  findJudge(
    3,
    (trust = [
      [1, 3],
      [2, 3],
    ])
  )
); //3

File: findJudge.py
class Solution:
    def findJudge(N, trust):
        if len(trust) < N-1:
            return -1
        inDegree = [0] * (N+1)
        outDegree = [0] * (N+1)
        for a, b in trust:
            outDegree[a] += 1
            inDegree[b] += 1
        for i in range(1, N+1):
            print(inDegree[i], outDegree[i])
            if inDegree[i] == N-1 and outDegree[i] == 0:
                return i
        return -1


print(Solution.findJudge(2, [[1, 2]]))  # 2
print(Solution.findJudge(3, [[1, 3], [2, 3]]))  # 3
print(Solution.findJudge(3, [[1, 3], [2, 3], [3, 1]]))  # -1
